### ISSUE TITLE
Allow Asciidoctor.js to be used in AsciidoctorJ

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctorj#189]
Allow Asciidoctor.js to be used as an alternate Asciidoctor implementation in AsciidoctorJ. The focus here is on hiding the implementation behind the Asciidoctor interface and allowing the Factory to select implementation when the Asciidoctor object is instantiated.

There's no need to support Rhino. We should focus on Nashorn and dynjs. Between the two, we shouldn't show a favorite in AsciidoctorJ.

**Comment by rmannibucau** [comment]
+1 for this, jruby stack has some issues in some env and js would be easy to support and a valid solution (workaround maybe but at least it "unblocks" on the user land)

**Comment by mojavelinux** [comment]
I believe that's what this project is all about: https://github.com/mogztter/asciidoctorg It's not integrated directly into AsciidoctorJ yet, but that is the plan, as I understand it.

**Comment by rmannibucau** [comment]
@mojavelinux my understanding is different, there are two steps in java/js integrations: 1. use a jsr223 ou v8 engine, 2. use graalvm which can have impacts on the JVM itself - it is not guaranteed that the graalvm new portability API works without running on graal itself AFAIK. Guess 1 is desired quickly, 2 is a nice to have for later IMHO.

**Comment by mojavelinux** [comment]
I could be wrong, but my understanding is that going forward, there is no other option for running JavaScript on the JVM than GraalVM because Nashorn is being removed (and Rhino is woefully outdated). Are you aware of another strategy?

**Comment by rmannibucau** [comment]
Yes, using v8 (https://github.com/eclipsesource/J2V8 - or https://github.com/flier/jav8), maybe one nashorn fork?, or even use frontend-maven-plugin core module to just run in node. I also confirm the project you mention does not work when not ran with graalvm so not that much an option yet :(.
